### ISSUE TITLE
All config parameters are stored in config.php and not in env.php

### ISSUE THREAD

**Issue by oneserv-heuser** [magmodules/magento2-channable#177]
We have four systems running our magento shop. We only want to enable and configure this extension on our production system.
But because all of the config parameters of this module are stored in the config.php, which is the same for all systems, we have to overwrite every single value in all of the env.php files.

I would appreciate it if all of the config parameters are shifted to env via the di.xml.
Example of how to achieve this:

```XML
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">

    <type name="Magento\Config\Model\Config\TypePool">
        <arguments>
            <argument name="environment" xsi:type="array">
                <item name="admin/url/use_custom" xsi:type="string">1</item>
            </argument>
        </arguments>
    </type>

</config>

```

**Comment by Marvin-Magmodules** [comment]
Hi @oneserv-heuser if you only want to enable/disable the module on all non production env, only flagging this option as an environment-specific setting would be sufficient right?

**Comment by oneserv-heuser** [comment]
That's right but I also want the product managers to be able to configure the module for themselves at the production system. If the configuration is present in the config.php they can't do this anymore as everything is grayed out (as documented [here](https://experienceleague.adobe.com/docs/commerce-operations/configuration-guide/cli/configuration-management/set-configuration-values.html#set-configuration-values-that-cannot-be-edited-in-the-admin))

So if now everything is placed inside the env.php I can omit the values inside the env.php for the production system and the product managers can configure everything as they want without me.

I looked into the configuration you have in this module, I think everything besides the payment method and shipping method configuration should be shifted, as all payment methods and shipping methods are handled in the config.php.

**Comment by Marvin-Magmodules** [comment]
@oneserv-heuser but if you would like them to edit these settings on production system, why are they all set in the config.php?

**Comment by oneserv-heuser** [comment]
@Marvin-Magmodules because every execution of `bin/magento app:config:dump` adds all config values to the config.php. Now I need to manually revert all the changes there every time I execute the command which happens after every new configuration option/new module, etc. because the configuration should be up to date on all systems.
If all the values are added to the env.php, it's not a problem as the env.php in my development environment is only used there and not on the production system.

**Comment by Marvin-Magmodules** [comment]
@oneserv-heuse, sure, let me dive into this next week. But how are you handeling this with core Magento settings? Looks like this is treated the same way?

**Comment by oneserv-heuser** [comment]
@Marvin-Magmodules Many of the core Magento settings are made to be the same across all systems. Unfortunately Magento has a lot of configuration settings that should be environment specific and not the same across all systems (e.g. e-mail templates). I have a module just for this with a huge di.xml file where i move everything I want from the config.php to the env.php

I'm really not into creating a ticket in Magento for these because it will take years to complete. Here I have the hope it will be slightly faster and I don't have to do it myself 😁

**Comment by Marvin-Magmodules** [comment]
@oneserv-heuser, sure I agree with you on e-mail templates in core, but looking at the Channable Module I'm not sure why we should add things like SKU, Title attribute selection, etc different across environments. I do agree we can add enable flags and web hook urls (and maybe some other fields) as these could indeed be environment-specific.

**Comment by oneserv-heuser** [comment]
@Marvin-Magmodules Yes, sure, some config options should stay in the config.php, didn't look through all specific possible settings.
I think you can better assess what to shift to the env.php, but enable flags and web hooks things should definitly be moved 👍🏻

**Comment by Marvin-Magmodules** [comment]
@oneserv-heuser great, will look into this coming week!

**Comment by Marvin-Magmodules** [comment]
Hi @oneserv-heuser,

We are happy to share that we've just released the new [v1.12.0](https://github.com/magmodules/magento2-channable/releases/tag/1.12.0) version, where we have added environment variables (see: https://github.com/magmodules/magento2-channable/commit/3116cd55de625dc053d9910f20c906c647ea67fd) We are closing this issue now but please feel free to reopen the issue if this still occurs.
